packages/databricks-sdk-js/src/api-client.ts

```diff
@@ -163,91 +152,38 @@ export class ApiClient {
+                if (!response.ok) {
+                    const err = parseErrorFromResponse(
+                        response.status,
+                        response.statusText,
+                        body
+                    );
+                    if (err.isRetryable()) {
                         throw new RetriableError();
-
-                    default:
-                        break;
+                    } else {
+                        throw logAndReturnError(
+                            url,
+                            options,
+                            response,
+                            err,
+                            context
+                        );
+                    }
                 }
-                return response;
+
+                return body;
             },
         });
 
-        let responseText: string;
-        try {
-            const responseBody = await response.arrayBuffer();
-            responseText = new TextDecoder().decode(responseBody);
-        } catch (e: any) {
-            logAndReturnError(url, options, "", e, context);
-            throw new ApiClientResponseError(
-                `Can't parse response from ${url.toString()}`,
-                ""
-            );
-        }
-
-        // throw error if the URL is incorrect and we get back an HTML page
-        if (response.headers.get("content-type")?.match("text/html")) {
-            // When the AAD tenant is not configured correctly, the response is a HTML page with a title like this:
-            // "Error 400 io.jsonwebtoken.IncorrectClaimException: Expected iss claim to be: https://sts.windows.net/aaaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaa/, but was: https://sts.windows.net/bbbbbbbbb-bbbb-bbbb-bbbb-bbbbbbbb/."
-            const m = responseText.match(/<title>(Error \d+.*?)<\/title>/);
-            let error: HttpError;
-            if (m) {
-                error = new HttpError(m[1], response.status);
-            } else {
-                error = new HttpError(
-                    `Can't connect to ${url.toString()}`,
-                    response.status
-                );
-            }
-
-            throw logAndReturnError(url, options, response, error, context);
-        }
-
-        // TODO proper error handling
-        if (!response.ok) {
-            const err = responseText.match(/invalid access token/i)
-                ? new HttpError("Invalid access token", response.status)
-                : new HttpError(responseText, response.status);
-            throw logAndReturnError(url, options, responseText, err, context);
-        }
-
         let responseJson: any;
         try {
             responseJson =
                 responseText.length === 0 ? {} : JSON.parse(responseText);
         } catch (e) {
             logAndReturnError(url, options, responseText, e, context);
-            throw new ApiClientResponseError(responseText, e);
+            throw new Error(`Can't parse reposne as JSON: ${responseText}`);
```

Comment:
We should send it through `parseError` as well.

Reply:
Let's unify this and https://github.com/databricks/databricks-vscode/pull/731/files#diff-dd9ade02b57b4295433e36501f1d12a708bd8bc95cdb04af866d851f5984691aR147-R151, 
into 1 error - IOError.